lib/elixir/lib/module.ex

```diff
@@ -801,14 +823,16 @@ defmodule Module do
   developer to customize it. See `Kernel.defoverridable/1` for
   more information and documentation.
   """
-  def make_overridable(module, tuples) do
+  @spec make_overridable(module, [function_arity]) :: :ok
+  def make_overridable(module, tuples) when is_atom(module) and is_list(tuples) do
     assert_not_compiled!(:make_overridable, module)
 
-    :lists.foreach(fn {name, arity} = tuple ->
+    :lists.foreach(fn {function_name, arity} = tuple
+        when is_atom(function_name) and is_integer(arity) and arity >= 0 and arity <= 255 ->
```

Comment:
I don't think this is a helpful error message. Blowing up with "no function clause matching" is gonna be pretty obscure since this comes from the guts of this function. I think here it could be beneficial to explicitly handle bad tuples:

```elixir
:lists.foreach(fn
  {function_name, arity} = tuple when is_atom(function_name) and is_integer(arity) and arity >= 0 and arity <= 255 ->
    ...
  other ->
    raise ArgumentError, "each element has to be a {function_name :: atom, arity :: 1..255} tuple, got: #{inspect(other)}"
end, ...)
```